src/WarpedPsf.cc

```diff
@@ -91,36 +93,56 @@ PTR(afw::detection::Psf::Image) warpAffine(
         );
     }
 
+    // allocate output image
+    afw::geom::Box2I bbox = computeBBoxFromTransform(im.getBBox(), t);
+    PTR(afw::detection::Psf::Image) ret = std::make_shared<afw::detection::Psf::Image>(bbox);
+
+    // zero-pad input image
+    PTR(afw::detection::Psf::Image) im_padded = zeroPadImage(im, xPad, yPad);
+
+    // warp it!
+    afw::math::warpImage(*ret, *im_padded, xyTransform, wc, 0.0);
+    return ret;
+}
+
+
+afw::geom::Box2I computeBBoxFromTransform
+    (afw::geom::Box2I const bbox,
+     afw::geom::AffineTransform const &t
+) {
+    static const double maxTransformCoeff = 200.0;
```

Comment:
Since you're checking against `maxTransformCoeff` here, I think you can probably delete the check in `warpAffine`.  Should probably bring the comment block down from there to here, though.